java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -2479,6 +2479,87 @@ public void zunion(BaseClient client) {
+        assertArrayEquals(
+                new String[] {"two"}, client.zinter(new KeyArray(new String[] {key1, key2})).get());
+        assertEquals(
+                Map.of("two", 5.5), client.zinterWithScores(new KeyArray(new String[] {key1, key2})).get());
+
+        // Intersection results are aggregated by the max score of elements
+        assertArrayEquals(
+                new String[] {"two"},
+                client.zinter(new KeyArray(new String[] {key1, key2}), Aggregate.MAX).get());
+        assertEquals(
+                Map.of("two", 3.5),
+                client.zinterWithScores(new KeyArray(new String[] {key1, key2}), Aggregate.MAX).get());
+
+        // Intersection results are aggregated by the min score of elements
+        assertArrayEquals(
+                new String[] {"two"},
+                client.zinter(new KeyArray(new String[] {key1, key2}), Aggregate.MIN).get());
+        assertEquals(
+                Map.of("two", 2.0),
+                client.zinterWithScores(new KeyArray(new String[] {key1, key2}), Aggregate.MIN).get());
+
+        // Intersection results are aggregated by the sum of the scores of elements
+        assertArrayEquals(
+                new String[] {"two"},
+                client.zinter(new KeyArray(new String[] {key1, key2}), Aggregate.SUM).get());
+        assertEquals(
+                Map.of("two", 5.5),
+                client.zinterWithScores(new KeyArray(new String[] {key1, key2}), Aggregate.SUM).get());
+
+        // Scores are multiplied by 2.0 for key1 and key2 during aggregation.
+        assertArrayEquals(
+                new String[] {"two"},
+                client.zinter(new WeightedKeys(List.of(Pair.of(key1, 2.0), Pair.of(key2, 2.0)))).get());
+        assertEquals(
+                Map.of("two", 11.0),
+                client
+                        .zinterWithScores(new WeightedKeys(List.of(Pair.of(key1, 2.0), Pair.of(key2, 2.0))))
+                        .get());
+
+        // Intersection results are aggregated by the minimum score,
+        // with scores for key1 multiplied by 1.0 and for key2 by -2.0.
+        assertArrayEquals(
+                new String[] {"two"},
+                client
+                        .zinter(
+                                new WeightedKeys(List.of(Pair.of(key1, 1.0), Pair.of(key2, -2.0))), Aggregate.MIN)
+                        .get());
+        assertEquals(
+                Map.of("two", -7.0),
+                client
+                        .zinterWithScores(
+                                new WeightedKeys(List.of(Pair.of(key1, 1.0), Pair.of(key2, -2.0))), Aggregate.MIN)
+                        .get());
+
+        // Non-existing Key - empty intersection
+        assertEquals(0, client.zinter(new KeyArray(new String[] {key1, key3})).get().length);
+        assertEquals(0, client.zinterWithScores(new KeyArray(new String[] {key1, key3})).get().size());
+
```

Comment:
Sho has been getting me to add tests that pass an empty key list, and asserting that a RequestException is thrown. Should we add that here? Or is it not possible to construct an empty key list with KeyArray or WeightedKeys?

Reply:
Added, thank you